info/v1/container.go

```diff
@@ -295,10 +295,24 @@ type CpuCFS struct {
 	ThrottledTime uint64 `json:"throttled_time"`
 }
 
-// All CPU usage metrics are cumulative from the creation of the container
+// Cpu Aggregated scheduler statistics
+type CpuSchedstat struct {
+	// https://www.kernel.org/doc/Documentation/scheduler/sched-stats.txt
+
+	// These numbers are summed up over all processes in a cgroup and can therefore not capture processes after they are gone.
+	// time spent on the cpu
+	RunTime uint64 `json:"run_time"`
+	// time spent waiting on a runqueue
+	RunqueueTime uint64 `json:"runqueue_time"`
+	// # of timeslices run on this cpu
+	RunPeriods uint64 `json:"run_periods"`
+}
+
+// CFS, Usage and LoadAverage metrics are cumulative from the creation of the container, Schedstat is the sum across the cumulative metrics of running processes
```

Comment:
How feasible would it be to maintain a cache of PID->schedstat metrics, update the PIDs that are active, and then sum over all PIDs ever seen?  This way we would get a cumulative metric, and we get the invariant that our reported schedstat metrics never decrease.

Reply:
Made a little something here: https://github.com/flix-tech/cadvisor/pull/1 So it is feasible, but there are of course still lost metrics between your housekeeping intervals.
I do not really like the code changes this requires.

Reply:
What in particular don't you like about the code changes?  I guess my main concern is that it is difficult to make sense of a number that decreases when a process dies.  I would rather have a slightly more complex implementation than a metric that isn't useful.  Most monitoring systems can't make sense of a total value that can decrease.

Reply:
I share your gripe that ContainerHandler contains so many fields it should not care about. Also I did not like the `seen`, but luckily that is gone.